protocol/linkpreview_unfurler_status.go

```diff
@@ -0,0 +1,194 @@
+package protocol
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/api/multiformat"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/communities"
+)
+
+type StatusUnfurler struct {
+	m      *Messenger
+	logger *zap.Logger
+	url    string
+}
+
+func NewStatusUnfurler(URL string, messenger *Messenger, logger *zap.Logger) *StatusUnfurler {
+	return &StatusUnfurler{
+		m:      messenger,
+		logger: logger,
+		url:    URL,
+	}
+}
+
+func buildThumbnail(image *images.IdentityImage, thumbnail *common.LinkPreviewThumbnail) error {
+	if image.IsEmpty() {
+		return nil
+	}
+
+	var err error
+
+	thumbnail.Width, thumbnail.Height, err = images.GetImageDimensions(image.Payload)
+	if err != nil {
+		return fmt.Errorf("failed to get image dimensions: %w", err)
+	}
+
+	thumbnail.DataURI, err = image.GetDataURI()
+	if err != nil {
+		return fmt.Errorf("failed to get data uri: %w", err)
+	}
+
+	return nil
+}
+
+func (u *StatusUnfurler) buildContactData(contactData *ContactURLData) (*common.StatusContactLinkPreview, error) {
+	c := new(common.StatusContactLinkPreview)
+	c.PublicKey = contactData.PublicKey
+	c.DisplayName = contactData.DisplayName
+	c.Description = contactData.Description
+
+	contactID, err := multiformat.DeserializeCompressedKey(contactData.PublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	contact := u.m.GetContactByID(contactID)
+	if contact == nil {
+		return c, nil
+	}
+
```

Comment:
if we got a contact object, we can update DisplayName and Description

Reply:
This is a good point!

@John-44, it's probably a question for you.
1. When unfurling a Status URL, we first have the data (`PublicKey`, `DisplayName`, `Description`) from the the URL itself.
2. Then we lookup the database for the such contact. If found, we also add the contact icon to the link preview.

But as @MishkaRogachev pointed out, we are able to update the `DisplayName` and `Description` with database values.

pros: 
-  most _probably_ database values are more up-to-date 
cons:
- the URL in the message text will still contain the old data, so there will be an inconsistency
- (I think) we don't have a way to distinguish which values are newer

WDYT?

Reply:
I've discussed this with Space Assasians crew. Here's the decision for unfurling status links:
- ignore the URL-encoded data (except IDs)
- take data from database (95% cases it will be there, because link is generated locally)
- if no data yet, fetch it from waku

I will correspondingly update this PR.

Thanks for bringing this up, @MishkaRogachev!